idb/postgres/postgres.go

```diff
@@ -2495,3 +2495,58 @@ func (db *IndexerDb) SetNetworkState(genesis bookkeeping.Genesis) error {
 	}
 	return db.setNetworkState(nil, &networkState)
 }
+
+// DeleteTransactions removes old transactions
+// keep is the number of rounds to keep in db
+func (db *IndexerDb) DeleteTransactions(ctx context.Context, keep uint64, timeout time.Duration) (int64, error) {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
```

Comment:
I don't think it makes sense to have a timeout as part of the interface here, the caller can pass in a timeout context if they want.